### PR TITLE
Remove CircleCI configuration

### PR DESCRIPTION
**Pull request by jasonrudolph**
**tl;dr** - Now that we're using VSTS for macOS builds, this pull request removes the CircleCI 1.0 build configuration before it is [EOL at the end of the month](https://circleci.com/blog/sunsetting-1-0/).

---

Atom has historically used CircleCI for macOS builds. CircleCI [announced](https://circleci.com/blog/sunsetting-1-0/) that CircleCI 1.0 builds will no longer be available after August 31, 2018. That means that we either need to migrate to CircleCI 2.0 or find another solution for Atom's macOS builds.

Given that we recently implemented cross-platform builds (macOS, Linux, and Windows) on VSTS for Atom Nightly Releases (https://github.com/atom/atom/pull/17445, https://github.com/atom/atom/pull/17538), and given that we don't yet have a working CircleCI 2.0 configuration (https://github.com/atom/atom/pull/17445), it feels like the pragmatic choice is to use the already-working VSTS builds as our definitive macOS CI solution, and remove our soon-to-be-EOL CircleCI configuration.

### CODE REVIEW

**Review by daviwil — APPROVED** [review]
:tada:
